poolmgr/gp.go

```diff
@@ -316,7 +319,13 @@ func (gp *GenericPool) specializePod(pod *v1.Pod, metadata *api.ObjectMeta) erro
 	// retry the specialize call a few times in case the env server hasn't come up yet
 	maxRetries := 20
 	for i := 0; i < maxRetries; i++ {
-		resp2, err := http.Post(specializeUrl, "text/plain", bytes.NewReader([]byte{}))
+
+		md, err := json.Marshal(metadata)
+		if err != nil {
+			return err
+		}
+
+		resp2, err := http.Post(specializeUrl, "application/json", bytes.NewReader(md))
```

Comment:
Why does the specialize endpoint need  function metadata? Does the workflow engine itself talk to the fission API to get the function? 

Can you load the workflow from a file instead? If so, #308 has everything you'll need.